src/s3/util/fileViewerManager.ts

```diff
@@ -65,13 +127,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation, fileNode)
         await newTab.openFileInReadOnly()
 
         this.activeTabs.set(fileLocation.fsPath, newTab)
     }
 
     public async openInEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (this.promptOnEdit) {
+            const message = localize(
+                'AWS.s3.fileViewer.warning.editStateWarning',
+                'Opening S3 file for editing. Saved changes will be written directly to the original location in S3.'
+            )
+
+            const dontShow = localize('AWS.s3.fileViewer.button.dismiss', "Don't show this again")
+            const help = localize('AWS.s3.fileViewer.button.help', 'Help')
+
+            this.window.showWarningMessage(message, dontShow, help).then(selection => {
+                if (selection === dontShow) {
+                    this.promptOnEdit = false
+                }
+
+                if (selection === help) {
+                    //add help section
```

Comment:
This will just be a link right? See `vscode.env.openExternal`

Reply:
this will be implemented after having a launch date for this feature